arch/x86/core/fatal.c

```diff
@@ -67,18 +68,20 @@ bool z_x86_check_stack_bounds(uintptr_t addr, size_t size, uint16_t cs)
 		start = (uintptr_t)Z_KERNEL_STACK_BUFFER(
 		    z_interrupt_stacks[cpu_id]);
 		end = start + CONFIG_ISR_STACK_SIZE;
-	} else if ((cs & 0x3U) != 0U ||
-		   (_current->base.user_options & K_USER) == 0) {
-		/* Thread was in user mode, or is not a user mode thread.
-		 * The normal stack buffer is what we will check.
+#ifdef CONFIG_USERSPACE
+	} else if ((cs & 0x3U) == 0 &&
```

Comment:
Pedantic, but might be clearer with a "IS_USERSPACE_CODE_SEG()" predicate or somesuch.

Reply:
agreed, at least a comment to make it immediately obvious what this check is for.

Reply:
added extra comments below this line.